### ISSUE TITLE
ThinkPad T480 Thunderbolt controller not found

### ISSUE THREAD

**Issue by xvitaly** [fwupd/fwupd#5975]
**Describe the bug**
Trying to install T480 Thunderbolt controller update, but fwupd fails.

**Steps to Reproduce**
```
$ fwupdmgr get-upgrades
Devices with no available firmware updates: 
 • MZVLW256HEHP-000L7
 • TPM
 • Thunderbolt Controller
 • Thunderbolt Controller
 • Thunderbolt Controller
 • Thunderbolt host controller
Devices with the latest available firmware version:
 • Embedded Controller
 • Intel Management Engine
 • System Firmware
```

```
$ wget https://fwupd.org/downloads/391c6be40422dc1d253f5a6b9bdf0c122ddc5f61f9fcb3ba4e61cf42a7c6f7bc-Lenovo-ThinkPad-T480-SystemFirmware-23.cab
$ fwupdmgr install ~/391c6be40422dc1d253f5a6b9bdf0c122ddc5f61f9fcb3ba4e61cf42a7c6f7bc-Lenovo-ThinkPad-T480-SystemFirmware-23.cab
Decompressing…           [ -                                     ]
No supported devices found
```

**Expected behavior**
The update should be installed.

**fwupd version information**
Please provide the version of the daemon and client.

```shell
$ fwupdmgr --version
compile   org.freedesktop.fwupd         1.8.14
compile   com.hughsie.libxmlb           0.3.11
compile   com.hughsie.libjcat           0.1.13
runtime   org.freedesktop.fwupd-efi     1.4
compile   org.freedesktop.gusb          0.4.5
runtime   com.hughsie.libjcat           0.1.13
runtime   com.dell.libsmbios            2.4
runtime   org.freedesktop.gusb          0.4.5
runtime   org.freedesktop.fwupd         1.8.14
runtime   org.kernel                    6.2.10-200.fc37.x86_64
```

Please note how you installed it (`apt`, `dnf`, `pacman`, source, etc):

Installed from Fedora 37 repository.

<summary>**fwupd device information**</summary>

Please provide the output of the fwupd devices recognized in your system.

```shell
LENOVO 20L50007RT
│
├─Core™ i7-8550U CPU @ 1.80GHz:
│     Device ID:          4bde70ba4e39b28f9eab1628f9dd6e6244c03027
│     Current version:    0x000000f0
│     Vendor:             Intel
│     GUIDs:              b9a2dd81-159e-5537-a7db-e7101d164d3f ← cpu
│                         30249f37-d140-5d3e-9319-186b1bd5cac3 ← CPUID\PRO_0&FAM_06
│                         561403e8-143a-5071-ab09-bf5e4c146983 ← CPUID\PRO_0&FAM_06&MOD_8E
│                         54bdc429-0179-5019-9cb6-8a343324c97c ← CPUID\PRO_0&FAM_06&MOD_8E&STP_A
│     Device Flags:       • Internal device
│   
├─Embedded Controller:
│     Device ID:          2292ae5236790b47884e37cf162dcf23bfcd1c60
│     Summary:            UEFI ESRT device
│     Current version:    0.1.22
│     Minimum Version:    0.1.22
│     Vendor:             Lenovo (DMI:LENOVO)
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               b87a926d-189c-49a1-b1ff-921099b1de89
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Supported on remote server
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│   
├─Intel Management Engine:
│     Device ID:          349bb341230b1a86e5effe7dfe4337e1590227bd
│     Summary:            UEFI ESRT device
│     Current version:    184.93.4323
│     Minimum Version:    0.0.1
│     Vendor:             Lenovo (DMI:LENOVO)
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               9e21f98b-fe98-455c-b388-da5450ab6979
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Supported on remote server
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│   
├─MZVLW256HEHP-000L7:
│     Device ID:          f2759da7fe8e0388c5f3601cb072f837b1070b03
│     Summary:            NVM Express solid state drive
│     Current version:    5L7QCXB7
│     Vendor:             Samsung (NVME:0x144D)
│     Serial Number:      S35ENA0K479749
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUIDs:              5b3df2da-f745-5fd0-81de-5dafd7f0bf8c ← NVME\VEN_144D&DEV_A804
│                         f87b9ac8-1cb3-5c0a-ae57-7144f211fe5e ← NVME\VEN_144D&DEV_A804&REV_00
│                         aed4d3c0-fd97-5e46-a32f-ff35e0692f6d ← NVME\VEN_144D&DEV_A804&SUBSYS_144DA801
│                         030c853f-259a-57a0-b3fb-1c66100db94b ← NVME\VEN_144D&DEV_A804&SUBSYS_144DA801&REV_00
│                         257d6faa-82f3-53e2-afa4-f6b9adc9595d ← SAMSUNG MZVLW256HEHP-000L7
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│                         • Signed Payload
│   
├─System Firmware:
│ │   Device ID:          a45df35ac0e948ee180fe216a5f703f32dda163f
│ │   Summary:            UEFI ESRT device
│ │   Current version:    0.1.47
│ │   Minimum Version:    0.1.0
│ │   Vendor:             Lenovo (DMI:LENOVO)
│ │   Update State:       Success
│ │   Problems:           [31m[1m• Device requires AC power to be connected[0m
│ │   GUIDs:              5d30ad91-8236-4c40-8249-ff61b38dae47
│ │                       230c8b18-8d9b-53ec-838b-6cfc0383493a ← main-system-firmware
│ │   Device Flags:       • Internal device
│ │                       • System requires external power source
│ │                       • Supported on remote server
│ │                       • Needs a reboot after installation
│ │                       • Cryptographic hash verification is available
│ │                       • Device is usable for the duration of the update
│ │                       • Updatable
│ │ 
│ ├─UEFI Platform Key:
│ │     Device ID:        6924110cde4fa051bfdc600a60620dc7aa9d3c6a
│ │     Summary:          Lenovo Ltd. PK CA 2012
│ │     Vendor:           Lenovo Ltd.
│ │     GUID:             71599d14-9b31-5270-b3bd-74c494585820 ← UEFI\CRT_9AEF2123F4DE7C19AFABD909BB2C8CAC4411E07E
│ │   
│ └─UEFI dbx:
│       Device ID:        362301da643102b9f38477387e2193e57abaa590
│       Summary:          UEFI revocation database
│       Current version:  217
│       Minimum Version:  217
│       Vendor:           UEFI:Linux Foundation
│       Install Duration: 1 second
│       GUIDs:            14503b3d-73ce-5d06-8137-77c68972a341 ← UEFI\CRT_A9087D1044AD18F7A94916D284CBC01827CF23CD8F60B79072C9CAA1FEF4D649
│                         5971a208-da00-5fce-b5f5-1234342f9cf7 ← UEFI\CRT_A9087D1044AD18F7A94916D284CBC01827CF23CD8F60B79072C9CAA1FEF4D649&ARCH_X64
│                         c6682ade-b5ec-57c4-b687-676351208742 ← UEFI\CRT_A1117F516A32CEFCBA3F2D1ACE10A87972FD6BBE8FE0D0B996E09E65D802A503
│                         f8ba2887-9411-5c36-9cee-88995bb39731 ← UEFI\CRT_A1117F516A32CEFCBA3F2D1ACE10A87972FD6BBE8FE0D0B996E09E65D802A503&ARCH_X64
│       Device Flags:     • Internal device
│                         • Updatable
│                         • Supported on remote server
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Only version upgrades are allowed
│                         • Signed Payload
│     
├─TPM:
│     Device ID:          c6a80ac3a22083423992a3cb15018989f37834d6
│     Current version:    73.4.17568.4452
│     Vendor:             ST Microelectronics (TPM:STM)
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUIDs:              ff71992e-52f7-5eea-94ef-883e56e034c6 ← system-tpm
│                         84df3581-f896-54d2-bd1a-372602f04c32 ← TPM\VEN_STM&DEV_0001
│                         bfaed10a-bbc1-525b-a329-35da2f63e918 ← TPM\VEN_STM&MOD_
│                         70b7b833-7e1a-550a-a291-b94a12d0f319 ← TPM\VEN_STM&DEV_0001&VER_2.0
│                         06f005e9-cb62-5d1a-82d9-13c534c53c48 ← TPM\VEN_STM&MOD_&VER_2.0
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device can recover flash failures
│                         • Updatable
│                         • Full disk encryption secrets may be invalidated when updating
│                         • Signed Payload
│   
├─Thunderbolt Controller:
│     Device ID:          f95c9218acd12697af946874bfe4239587209232
│     Summary:            UEFI ESRT device
│     Current version:    258.1081
│     Minimum Version:    0.1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               24aac206-3921-45c0-a79e-0a607b11aef0
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│   
├─Thunderbolt Controller:
│     Device ID:          d96de5c124b60ed6241ebcb6bb2c839cb5580786
│     Summary:            UEFI ESRT device
│     Current version:    258.1081
│     Minimum Version:    0.1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               77e3677e-dcc2-465c-b2a0-6c37aa6ba300
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│   
├─Thunderbolt Controller:
│     Device ID:          f37fb01122dd62c773f4e84ec89737e059712d59
│     Summary:            UEFI ESRT device
│     Current version:    73.4
│     Minimum Version:    73.4
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               e0049080-09bd-473a-a02c-19aa0c8dad78
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│   
├─Thunderbolt host controller:
│     Device ID:          c994c96d4262316c36825e4480dc79e618baa1e7
│     Summary:            Unmatched performance for high-speed I/O
│     Current version:    20.00
│     Vendor:             Lenovo (THUNDERBOLT:0x0109, TBT:0x0109)
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUIDs:              6d4d771c-0579-504d-92a1-737180c7b360 ← THUNDERBOLT\VEN_0109&DEV_1707
│                         8487e4cb-fe6b-5cb6-ad88-46535fb110e1 ← THUNDERBOLT\VEN_0109&DEV_1707&REV_00
│                         7d131734-64ed-5865-a46e-a67cfb9b5a46 ← TBT-01091707
│                         e45313fd-8d5c-54e9-b175-1f15ffe6b05c ← TBT-01091707-controller0-0
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Device stages updates
│                         • Updatable
│                         • Signed Payload
│   
└─UHD Graphics 620:
      Device ID:          5792b48846ce271fab11c4a545f7a3df0d36e00a
      Current version:    07
      Vendor:             Intel Corporation (PCI:0x8086)
      GUIDs:              8b72a10c-1279-5f8e-a28a-34fb11a58240 ← PCI\VEN_8086&DEV_5917
                          fed2efa4-7045-55a1-b4fc-b29283d59fe5 ← PCI\VEN_8086&DEV_5917&REV_07
                          a52425ff-2c89-5aef-b6b1-34bc70fb5c99 ← PCI\VEN_8086&DEV_5917&SUBSYS_17AA225D
                          b751f9df-b238-53fd-b30b-ce22116f8b39 ← PCI\VEN_8086&DEV_5917&SUBSYS_17AA225D&REV_07
                          8d939522-f999-5920-9024-e75504c6867b ← PCI\VEN_8086&DEV_5917&REV_00
                          8cc739bb-cc29-5696-9f47-ac970b7f4deb ← PCI\VEN_8086&DEV_5917&SUBSYS_17AA225D&REV_00
      Device Flags:       • Internal device
                          • Cryptographic hash verification is available
```

</details>

**System UEFI configuration**
Please provide the output of the following commands:

```shell
BootCurrent: 0000
Timeout: 2 seconds
BootOrder: 0000,0017,0018,0019,001A,001B,001C,001D,001E,001F,0001
Boot0000* Fedora	HD(1,GPT,185d3e7c-f253-4004-85e9-eeb66fa23ca8,0x800,0x64000)/File(\EFI\fedora\shimx64.efi)
      dp: 04 01 2a 00 01 00 00 00 00 08 00 00 00 00 00 00 00 40 06 00 00 00 00 00 7c 3e 5d 18 53 f2 04 40 85 e9 ee b6 6f a2 3c a8 02 02 / 04 04 34 00 5c 00 45 00 46 00 49 00 5c 00 66 00 65 00 64 00 6f 00 72 00 61 00 5c 00 73 00 68 00 69 00 6d 00 78 00 36 00 34 00 2e 00 65 00 66 00 69 00 00 00 / 7f ff 04 00
Boot0001* Linux-Firmware-Updater	HD(1,GPT,185d3e7c-f253-4004-85e9-eeb66fa23ca8,0x800,0x64000)/File(\EFI\fedora\shimx64.efi) File(.\fwupdx64.efi)
      dp: 04 01 2a 00 01 00 00 00 00 08 00 00 00 00 00 00 00 40 06 00 00 00 00 00 7c 3e 5d 18 53 f2 04 40 85 e9 ee b6 6f a2 3c a8 02 02 / 04 04 34 00 5c 00 45 00 46 00 49 00 5c 00 66 00 65 00 64 00 6f 00 72 00 61 00 5c 00 73 00 68 00 69 00 6d 00 78 00 36 00 34 00 2e 00 65 00 66 00 69 00 00 00 / 7f ff 04 00
    data: 5c 00 66 00 77 00 75 00 70 00 64 00 78 00 36 00 34 00 2e 00 65 00 66 00 69 00 00 00
Boot0010  Setup	FvFile(721c8b66-426c-4e86-8e99-3457c46ab0b9)
      dp: 04 06 14 00 66 8b 1c 72 6c 42 86 4e 8e 99 34 57 c4 6a b0 b9 / 7f ff 04 00
Boot0011  Boot Menu	FvFile(126a762d-5758-4fca-8531-201a7f57f850)
      dp: 04 06 14 00 2d 76 6a 12 58 57 ca 4f 85 31 20 1a 7f 57 f8 50 / 7f ff 04 00
Boot0012  Diagnostic Splash Screen	FvFile(a7d8d9a6-6ab0-4aeb-ad9d-163e59a7a380)
      dp: 04 06 14 00 a6 d9 d8 a7 b0 6a eb 4a ad 9d 16 3e 59 a7 a3 80 / 7f ff 04 00
Boot0013  Lenovo Diagnostics	FvFile(3f7e615b-0d45-4f80-88dc-26b234958560)
      dp: 04 06 14 00 5b 61 7e 3f 45 0d 80 4f 88 dc 26 b2 34 95 85 60 / 7f ff 04 00
Boot0014  Startup Interrupt Menu	FvFile(f46ee6f4-4785-43a3-923d-7f786c3c8479)
      dp: 04 06 14 00 f4 e6 6e f4 85 47 a3 43 92 3d 7f 78 6c 3c 84 79 / 7f ff 04 00
Boot0015  Rescue and Recovery	FvFile(665d3f60-ad3e-4cad-8e26-db46eee9f1b5)
      dp: 04 06 14 00 60 3f 5d 66 3e ad ad 4c 8e 26 db 46 ee e9 f1 b5 / 7f ff 04 00
Boot0016  MEBx Hot Key	FvFile(ac6fd56a-3d41-4efd-a1b9-870293811a28)
      dp: 04 06 14 00 6a d5 6f ac 41 3d fd 4e a1 b9 87 02 93 81 1a 28 / 7f ff 04 00
Boot0017* USB CD	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,86701296aa5a7848b66cd49dd3ba6a55)
      dp: 03 0a 24 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 86 70 12 96 aa 5a 78 48 b6 6c d4 9d d3 ba 6a 55 / 7f ff 04 00
Boot0018* USB FDD	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,6ff015a28830b543a8b8641009461e49)
      dp: 03 0a 24 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 6f f0 15 a2 88 30 b5 43 a8 b8 64 10 09 46 1e 49 / 7f ff 04 00
Boot0019* NVMe1	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,001c199932d94c4eae9aa0b6e98eb8a401)
      dp: 03 0a 25 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 00 1c 19 99 32 d9 4c 4e ae 9a a0 b6 e9 8e b8 a4 01 / 7f ff 04 00
Boot001A* NVMe0	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,001c199932d94c4eae9aa0b6e98eb8a400)
      dp: 03 0a 25 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 00 1c 19 99 32 d9 4c 4e ae 9a a0 b6 e9 8e b8 a4 00 / 7f ff 04 00
Boot001B* ATA HDD0	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,91af625956449f41a7b91f4f892ab0f602)
      dp: 03 0a 25 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 91 af 62 59 56 44 9f 41 a7 b9 1f 4f 89 2a b0 f6 02 / 7f ff 04 00
Boot001C* USB HDD	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,33e821aaaf33bc4789bd419f88c50803)
      dp: 03 0a 24 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 33 e8 21 aa af 33 bc 47 89 bd 41 9f 88 c5 08 03 / 7f ff 04 00
Boot001D* PCI LAN	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,78a84aaf2b2afc4ea79cf5cc8f3d3803)
      dp: 03 0a 24 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 78 a8 4a af 2b 2a fc 4e a7 9c f5 cc 8f 3d 38 03 / 7f ff 04 00
Boot001E  Other CD	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,aea2090adfde214e8b3a5e471856a35406)
      dp: 03 0a 25 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b ae a2 09 0a df de 21 4e 8b 3a 5e 47 18 56 a3 54 06 / 7f ff 04 00
Boot001F  Other HDD	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,91af625956449f41a7b91f4f892ab0f606)
      dp: 03 0a 25 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 91 af 62 59 56 44 9f 41 a7 b9 1f 4f 89 2a b0 f6 06 / 7f ff 04 00
Boot0020* USBR BOOT CDROM	PciRoot(0x0)/Pci(0x14,0x0)/USB(11,1)
      dp: 02 01 0c 00 d0 41 03 0a 00 00 00 00 / 01 01 06 00 00 14 / 03 05 06 00 0b 01 / 7f ff 04 00
Boot0021* USBR BOOT Floppy	PciRoot(0x0)/Pci(0x14,0x0)/USB(11,0)
      dp: 02 01 0c 00 d0 41 03 0a 00 00 00 00 / 01 01 06 00 00 14 / 03 05 06 00 0b 00 / 7f ff 04 00
Boot0022* ATA HDD	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,91af625956449f41a7b91f4f892ab0f6)
      dp: 03 0a 24 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b 91 af 62 59 56 44 9f 41 a7 b9 1f 4f 89 2a b0 f6 / 7f ff 04 00
Boot0023* ATAPI CD	VenMsg(bc7838d2-0f82-4d60-8316-c068ee79d25b,aea2090adfde214e8b3a5e471856a354)
      dp: 03 0a 24 00 d2 38 78 bc 82 0f 60 4d 83 16 c0 68 ee 79 d2 5b ae a2 09 0a df de 21 4e 8b 3a 5e 47 18 56 a3 54 / 7f ff 04 00
```

```shell
$ efivar -l | grep fw
0abba7dc-e516-4167-bbf5-4d9d1c739416-fwupd-5d30ad91-8236-4c40-8249-ff61b38dae47-0
```

```shell
$ tree /boot
/boot
├── config-6.2.10-200.fc37.x86_64
├── config-6.2.9-200.fc37.x86_64
├── efi  [error opening dir]
├── grub2  [error opening dir]
├── initramfs-0-rescue-d81e28cb17d24c1e9176a83c32f34d23.img
├── initramfs-6.2.10-200.fc37.x86_64.img
├── initramfs-6.2.9-200.fc37.x86_64.img
├── loader
│   └── entries  [error opening dir]
├── lost+found  [error opening dir]
├── symvers-6.2.10-200.fc37.x86_64.gz -> /lib/modules/6.2.10-200.fc37.x86_64/symvers.gz
├── symvers-6.2.9-200.fc37.x86_64.gz -> /lib/modules/6.2.9-200.fc37.x86_64/symvers.gz
├── System.map-6.2.10-200.fc37.x86_64
├── System.map-6.2.9-200.fc37.x86_64
├── vmlinuz-0-rescue-d81e28cb17d24c1e9176a83c32f34d23
├── vmlinuz-6.2.10-200.fc37.x86_64
└── vmlinuz-6.2.9-200.fc37.x86_64

6 directories, 12 files
```

**Additional questions**

- Operating system and version: Fedora 37
- Have you tried rebooting? Yes
- Is this a regression? Maybe
- Are you using an NVMe disk? yes
- Is secure boot enabled? Yes
- Is this a Lenovo system with 'Boot Order Lock' turned on in the BIOS? No

**Comment by hughsie** [comment]
@anttih can you try https://github.com/fwupd/fwupd/pull/5976 please? If it helps I can do a Fedora RPM.

**Comment by anttih** [comment]
@hughsie I quickly tried but can't install it due to some of the [Nix patches](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/os-specific/linux/firmware/fwupd/default.nix#L140) not working for the `master` version of fwupd. I'll see if I can make a workaround.

**Comment by anttih** [comment]
I backported the patch and it works! I was able to update the Thunderbolt firmware from 14.0 -> 23.0.

```
├─Thunderbolt host controller:
│     Device ID:          d3c506436c4bb90704c5837177b7da616901aabd
│     Summary:            Unmatched performance for high-speed I/O
│     Current version:    23.00
│     Vendor:             Lenovo (THUNDERBOLT:0x0109, TBT:0x0109)
│     Update State:       Success
│     GUIDs:              6d4d771c-0579-504d-92a1-737180c7b360 ← THUNDERBOLT\VEN_0109&DEV_1707
│                         8487e4cb-fe6b-5cb6-ad88-46535fb110e1 ← THUNDERBOLT\VEN_0109&DEV_1707&REV_00
│                         fe328c2d-5300-597b-a21d-991116db8ab6 ← TBT-01091707-native
│                         15a5ae1d-0ec9-5013-9221-cd6dd753fcb0 ← TBT-01091707-native-controller0-0
│
```

**Comment by xvitaly** [comment]
@hughsie Can you build a Fedora RPM scratch build with this patch? I can test.

**Comment by hughsie** [comment]
Can you try https://copr.fedorainfracloud.org/coprs/rhughes/fwupd/ please

**Comment by mrhpearson** [comment]
Thanks to all for their patience while we tracked this down and mega-thanks to @hughsie for figuring out the fix - very awesome!

**Comment by xvitaly** [comment]
> Can you try https://copr.fedorainfracloud.org/coprs/rhughes/fwupd/ please

It works. Thanks.

**Comment by hughsie** [comment]
Excellent, thanks. I've backported it to 1_8_X too.

**Comment by xvitaly** [comment]
@hughsie Can you push 1.8.17 to Fedora 37 repositories please?

**Comment by hughsie** [comment]
Mea culpa; building now. I'll submit the update in ~1h.

**Comment by xvitaly** [comment]
> Mea culpa; building now. I'll submit the update in ~1h.

Thank you.

**Comment by hughsie** [comment]
https://bodhi.fedoraproject.org/updates/FEDORA-2023-ed2793b658

**Comment by TheMrRandomDude** [comment]
Hey people. What if fwupdmgr get-devices or get-upgrades (after refresh), (tried both as sudo and without,) isn't finding "Thunderbolt Controller" at all. None in list. I did apt update & upgrade beforehand too.

**Comment by christal87** [comment]
I've came across this closed issue today while updating  the tbolt fw on my T480. Debian 12.4.0 has an older version (1.8.12) packed, which I suspect was affected. So I cloned the main branch, set up a test virtualenv as per the readme, shut down the older systemd service as a temporary measure and updated the fw successfully from 14 to 23. I'd like to leave this here to anyone as a guideline. Thanks guys for your hard work!

**Comment by va1entin** [comment]
I kept having this issue on my T480 with fwupd `1.9.21` on Arch with TB assist mode and secure boot disabled in BIOS. I also reset the BIOS, making sure TB assist mode and secure boot were still disabled, to no avail: No "Thunderbolt host controller" device would be visible whatever I tried.

I managed to fix it by setting Thunderbolt `Security level` to `User Authorization` in BIOS as per @xvitaly's screenshot from earlier. It was set to `Display Port and USB` on my machine for some reason - even after BIOS reset.
After changing the security level the Thunderbolt host controller appeared in fwupdmgr and I was able to update FW from 21 to 23.

Hope this might help someone in the future!

> > [@xvitaly](https://github.com/xvitaly) , this GUID seems the assist mode is still enabled, the FW team requested to enter F10 to exit BIOS after disabled assist mode. Could you check this? thanks
> 
> ![aaaa](https://github.com/fwupd/firmware-lenovo/assets/966711/5670013e-adf9-4715-a04a-a6ad323e1dc5)
